src/EnergyPlus/PhotovoltaicThermalCollectors.cc

```diff
@@ -165,60 +168,96 @@ namespace PhotovoltaicThermalCollectors {
+        if (NumBIPVTPerform > 0) GetBIPVTCollectorsInput(state, NumBIPVTPerform, tmpBIPVTperf);
 
         // now get main PVT objects
         state.dataIPShortCut->cCurrentModuleObject = "SolarCollector:FlatPlate:PhotovoltaicThermal";
         state.dataPhotovoltaicThermalCollector->NumPVT =
             state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, state.dataIPShortCut->cCurrentModuleObject);
-        state.dataPhotovoltaicThermalCollector->PVT.allocate(state.dataPhotovoltaicThermalCollector->NumPVT);
+        if (state.dataPhotovoltaicThermalCollector->NumPVT > 0)
+            GetMainPVTInput(
+                state, state.dataPhotovoltaicThermalCollector->NumPVT, state.dataPhotovoltaicThermalCollector->PVT, tmpSimplePVTperf, tmpBIPVTperf);
+        if (allocated(tmpSimplePVTperf)) tmpSimplePVTperf.deallocate();
+        if (allocated(tmpBIPVTperf)) tmpBIPVTperf.deallocate();
+    }
+
+    void GetPVTSimpleCollectorsInput(EnergyPlusData &state, int NumSimplePVTPerform, Array1D<SimplePVTModelStruct> &tmpSimplePVTperf)
+    {
+        // PURPOSE OF THIS SUBROUTINE:
+        // Get input for PVT Simple objects
+
+        int Item;                // Item to be "gotten"
+        int NumAlphas;           // Number of Alphas for each GetObjectItem call
+        int NumNumbers;          // Number of Numbers for each GetObjectItem call
+        int IOStatus;            // Used in GetObjectItem
+        bool ErrorsFound(false); // Set to true if errors in input, fatal at end of routine
+        using ScheduleManager::GetScheduleIndex;
+
+        tmpSimplePVTperf.allocate(NumSimplePVTPerform);
+        for (Item = 1; Item <= NumSimplePVTPerform; ++Item) {
+            state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                                     state.dataIPShortCut->cCurrentModuleObject,
+                                                                     Item,
+                                                                     state.dataIPShortCut->cAlphaArgs,
+                                                                     NumAlphas,
+                                                                     state.dataIPShortCut->rNumericArgs,
+                                                                     NumNumbers,
+                                                                     IOStatus,
+                                                                     _,
+                                                                     state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                     state.dataIPShortCut->cAlphaFieldNames,
+                                                                     state.dataIPShortCut->cNumericFieldNames);
+            if (UtilityRoutines::IsNameEmpty(state, state.dataIPShortCut->cAlphaArgs(1), state.dataIPShortCut->cCurrentModuleObject, ErrorsFound))
```

Comment:
This check is not necessary. This gets caught by the epJSON parser, which prints an error and exits. If you got here, the name is not empty.

Reply:
removed